lib/rules/no-restricted-imports.js

```diff
@@ -121,31 +139,125 @@ module.exports = {
         }
 
         /**
+         * Report a restricted path specifically for patterns.
+         * @param {node} node - representing the restricted path reference
+         * @returns {void}
+         * @private
+         */
+        function reportPathForPatterns(node) {
+            const importName = node.source.value.trim();
+
+            context.report({
+                node,
+                message: "'{{importName}}' import is restricted from being used by a pattern.",
+                data: {
+                    importName
+                }
+            });
+        }
+
+        /**
+         * Report a restricted path specifically for patterns.
+         * @param {string} name - path of the import
+         * @param {node} node - representing the restricted path reference
+         * @returns {void}
+         * @private
+         */
+        function reportPathForEverythingImported(name, node) {
+            const importName = node.source.value.trim();
+            const importObject = restrictedPathMessages[name].objectNames;
+
+            context.report({
+                node,
+                message: "You cannot import * because '{{importObject}}' from '{{importName}}' is restricted.",
```

Comment:
I would make this message less personal. I can totally see somebody reading that message in their editor and yelling "You don't tell me what I can or cannot do!". Something like "* import is not valid because '{{importObject}}' from '{{importName}}' is restricted.".

Reply:
Ha! That's definitely true. I've updated the message 👍